packages/react-components/src/components/ChatMessage/ChatMessageContent.tsx

```diff
@@ -78,16 +80,18 @@ const MessageContentWithLiveAria = (props: MessageContentWithLiveAriaProps): JSX
 const MessageContentAsRichTextHTML = (props: ChatMessageContentProps): JSX.Element => {
   /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
   useEffect(() => {
-    props.message.attachedFilesMetadata?.map((fileMetadata) => {
-      if (
-        props.onFetchAttachment &&
-        props.attachmentsMap &&
-        fileMetadata.attachmentType === 'inlineImage' &&
-        props.attachmentsMap[fileMetadata.id] === undefined
-      ) {
-        props.onFetchAttachment(fileMetadata);
-      }
+    const attachments = props.message.attachedFilesMetadata?.filter((fileMetadata) => {
+      return fileMetadata.attachmentType === 'inlineImage';
     });
+
+    if (props.attachmentsMap && attachments) {
+      attachments.forEach((fileMetadata) => {
+        if (props.onFetchAttachment && props.attachmentsMap && props.attachmentsMap[fileMetadata.id] === undefined) {
+          props.onFetchAttachment([fileMetadata], props.message);
```

Comment:
It doesn't matter, its actually a bit more efficient this way because as long as a message is even edited the attachment id changes for all items. So if we detect any change at all we will need to refresh. I suppose if we wanted to be really efficient all we need to do was to check if the first item has change and if so make the full request. But IMO this is just slightly safer.